managers/backup-manager/BackupManager.js

```diff
@@ -0,0 +1,421 @@
+                    return this.backupStore
+                      .deleteBackupFile(options);
+                  }
+                })
+                .catch((err) => logger.error('Error occurred while performing clean up of backup failure operation : ', err));
+            }
+          }).then(() => {
+            throw err;
+          });
+      });
+  }
+
+  getOperationState(name, opts) {
+    logger.info(`Retrieving state of last Backup with:`, opts);
+    return this.getBackupOperationState(opts)
+      .then(result => {
+        const deploymentName = opts.deployment;
+        const action = _.capitalize(name);
+        const timestamp = result.updated_at;
+        switch (result.state) {
+        case CONST.APISERVER.RESOURCE_STATE.SUCCEEDED:
+          return {
+            description: `${action} deployment ${deploymentName} succeeded at ${timestamp}`,
+            state: CONST.APISERVER.RESOURCE_STATE.SUCCEEDED
+          };
+        case CONST.APISERVER.RESOURCE_STATE.ABORTED:
+          return {
+            description: `${action} deployment ${deploymentName} aborted at ${timestamp}`,
+            state: CONST.APISERVER.RESOURCE_STATE.FAILED
+          };
+        case CONST.APISERVER.RESOURCE_STATE.FAILED:
+          return {
+            description: `${action} deployment ${deploymentName} failed at ${timestamp} with Error "${result.stage}"`,
+            state: CONST.APISERVER.RESOURCE_STATE.FAILED
+          };
+        default:
+          return {
+            description: `${action} deployment ${deploymentName} is still in progress: "${result.stage}"`,
+            state: CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS
+          };
+        }
+      });
+  }
+
+  getBackupOperationState(opts) {
+    logger.info('Getting Backup operation State for:', opts);
+    const agent_ip = opts.agent_ip;
+    const options = _.assign({
+      service_id: this.plan.service.id,
+      plan_id: this.plan.id,
+      tenant_id: opts.context ? this.getTenantGuid(opts.context) : opts.tenant_id
+    }, opts);
+
+    function isFinished(state) {
+      return _.includes(['succeeded', 'failed', 'aborted'], state);
+    }
+
+    return this.agent
+      .getBackupLastOperation(agent_ip)
+      .tap(lastOperation => {
+        if (isFinished(lastOperation.state)) {
+          return this.agent
+            .getBackupLogs(agent_ip)
+            .tap(logs => _.each(logs, log => logger.info(`Backup log for: ${opts.instance_guid} - ${JSON.stringify(log)}`)))
+            .then(logs => this.backupStore
+              .patchBackupFile(options, {
+                state: lastOperation.state,
+                logs: logs,
+                snapshotId: lastOperation.snapshotId
+              })
+            ).then(() => this.backupStore.getBackupFile(options))
+            .then(metadata => eventmesh.apiServerClient.updateOperationResponse({
```

Comment:
Here just updating Operation Response, not updating operation state like how its done in start backup?